signac/common/config.py

```diff
@@ -99,45 +98,33 @@ def read_config_file(filename, configspec=None, *args, **kwargs):
     return config
 
 
-def load_config(root=None, local=False):
-    """Load configuration, searching upward from a root path if desired.
+def load_config(root=None):
+    """Load configuration at root directory.
 
     Parameters
     ----------
     root : str
-        The path from which to begin searching for config files.
-    local : bool, optional
-        If ``True``, only search in the provided directory and do not traverse
-        upwards through the filesystem (Default value: False).
+        The path from which to load the local config.
 
     Returns
     --------
     :class:`Config`
         The composite configuration including both local and global config data
-        if requested.
+        if requested. Note that because this config is a composite,
+        modifications to the returned value will not be reflected in the files.
     """
     if root is None:
         root = os.getcwd()
     config = Config(configspec=cfg.split("\n"))
```

Comment:
We need to rename `cfg` to indicate it's a module-level constant, like `CONFIG_SCHEMA` or similar. It took me a moment to recognize what was going on here.

Reply:
Agreed. It is a public attribute, though, which isn't something that's really feasible to deprecate with a warning. I think we'll just have to rename this in 2.0. Chances are nobody is using it, but I know signac-flow used to (I believe that I removed that dependence recently when I removed monkey-patching), which is why I'm advocating treating it as a breaking change for 2.0 rather than something to put in this PR.

Reply:
Again, forgot this was a PR where breaks were fine. That said, I think this is another future TODO IMO.